packages/types/src/integration.ts

```diff
@@ -13,6 +13,48 @@ export interface IntegrationClass<T> {
   new (...args: any[]): T;
 }
 
+/**
+ * An integration in function form.
+ * This is expected to return an integration result,
+ * and also have a `name` property that holds the integration name (so we can e.g. filter these before actually calling them).
+ */
+export type IntegrationFn<Fn extends (...rest: any[]) => IntegrationFnResult> = { id: string } & Fn;
+
+export interface IntegrationFnResult {
```

Comment:
I don't think these are necessary or a good idea. I would prefer if we just have an `Integration` interface. These types seem overkill and don't add much.

Reply:
The thing is that for now these are _not_ the same, as `setupOnce` is optional here. In v8 we can get rid of this and have just a single `Integration` interface again!